miden-lib/asm/kernels/transaction/api.masm

```diff
@@ -466,19 +466,23 @@ export.get_block_hash
     # => [BLOCK_HASH]
 end
 
-#! Returns the input notes hash. This is computed as a sequential hash of (nullifier, script_root)
-#! tuples over all input notes.
+#! Returns the input notes commitment.
+#!
+#! This is computed as a sequential hash of `(NULLIFIER, ZERO_OR_NOTEID)` over all input notes. The
+#! data `ZERO_OR_NOTEID` functions as a flag, if the value is set to zero, then the notes are
+#! authenticated by the transaction kernel. If the value is non-zero, then note authentication will
+#! be delayed to the batch kernel. The delayed authentication allows a transaction to consume a
```

Comment:
nit: it could also be delayed to the block kernel. maybe here we say "batch/block kernels"?

Reply:
changed